linux/hci/socket/socket.go

```diff
@@ -131,5 +138,15 @@ func (s *Socket) Write(p []byte) (int, error) {
 }
 
 func (s *Socket) Close() error {
+	// Prevent subsequent Read() before we close the socket.
+	close(s.closed)
+
+	// Any pending reference to the socket prevents the underlying HCI device from being released.
+	// This most likely happens due to a blocked Read().
+	// The Close() syscall does not unblock the Read(), and the HCI socket does not support
+	// Shutdown() syscall, which can be used for a TCP socket.
+	// Here we just send a HCI Reset Command, which results a HCI response that subsequently unblocks the Read().
+	unix.Write(s.fd, []byte{0x01, 0x09, 0x10, 0x00})
+
```

Comment:
Could we lock the `wmu` and `rmu` here, to assure that the `Read()` and `Write()` are actually done and cannot be called during the closing of the socket?
Then `Close()` would wait until the pending `Read()` is done, then close the socket and return.